### PR TITLE
Fix: map layer permission issue

### PR DESCRIPTION
**Pull request by taylorn01**
Test:

- Create a new user. 
- Give them all the permissions needed to navigate the app. 
- Create an Activity and give it geometry data.
- Add this Activity to the Global Set.
- Add the Global Set to the Global Groups permissions.
- Create a Heritage Asset and give it geometry data.
- Create a Logical Set with this member definition `resource-type-filter=[{"graphid":"076f9381-7b00-11e9-8d6b-80000b44d1d9","name":"Heritage Asset","inverted":false}]`
- Add the Logical Set to the Global Groups permissions.
- Add the new user to the members of the Global Group.
- Run `make manage CMD="apply_sets"`
- Login in as the other user and test if they can see both map layers for Activity and Heritage Asset.
- After remove the Set or Logical Set to test they can't see it any more (re-run apply_sets)
- Remove both and check they also can't see that one as well

### CODE REVIEW

File: coral/permissions/casbin.py [hunk 1]
```diff
@@ -732,6 +732,13 @@ def get_restricted_instances(self, user, search_engine=None, allresources=False)
 
         if allresources is False and user.is_superuser is True:
             return []
+        
+        if allresources is True:
+            # Loop through casbin table for ids that match the user id and container resource instances
+            # -> Logical sets will have some sort of implementation here as well for gather rids
+            # Create array of rids the user has access to
```
Comment (philtweir) [inline coral/permissions/casbin.py:739]: Is 744-754 not sufficient to cover this case, @taylorn01 ? (in particular, if we can avoid hitting the DB, as opposed to ES, then it'll be faster)

Comment (philtweir) [reply coral/permissions/casbin.py]: Oh, I see - search_engine wasn't necessarily set, strange.

File: coral/permissions/casbin.py [hunk 2]
```diff
@@ -732,9 +732,30 @@ def get_restricted_instances(self, user, search_engine=None, allresources=False)
...
+        # Exclude set IDs the user DOES have access to to gather a list of resource
+        # IDs they do not have access to
+        query.add_query({
+            "bool": {
+                "must_not": {
+                    "nested": {
+                        "path": "sets",
+                        "query": {
+                            "terms": {
+                                "sets.id": user_sets
```
[the remark this answers is not in the document]
Comment (taylorn01) [reply coral/permissions/casbin.py:752]: _Didn't realise I left this comment in pending_

**Review by StuCM — APPROVED** [review]
Ran all the tests and its working.
Activity is present when added to the global set and the Heritage asset is visible with the additional Logical set and is again hidden when this is removed

**Pull request merged**